// @aws-sdk/client-ec2/dist-es/commands/AssociateTransitGatewayPolicyTableCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateTransitGatewayPolicyTableRequestFilterSensitiveLog, AssociateTransitGatewayPolicyTableResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_ec2AssociateTransitGatewayPolicyTableCommand, serializeAws_ec2AssociateTransitGatewayPolicyTableCommand, } from "../protocols/Aws_ec2";
var AssociateTransitGatewayPolicyTableCommand = (function (_super) {
    __extends(AssociateTransitGatewayPolicyTableCommand, _super);
    function AssociateTransitGatewayPolicyTableCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateTransitGatewayPolicyTableCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "AssociateTransitGatewayPolicyTableCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateTransitGatewayPolicyTableRequestFilterSensitiveLog,
            outputFilterSensitiveLog: AssociateTransitGatewayPolicyTableResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateTransitGatewayPolicyTableCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2AssociateTransitGatewayPolicyTableCommand(input, context);
    };
    AssociateTransitGatewayPolicyTableCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2AssociateTransitGatewayPolicyTableCommand(output, context);
    };
    return AssociateTransitGatewayPolicyTableCommand;
}($Command));
export { AssociateTransitGatewayPolicyTableCommand };
