// @aws-sdk/client-ec2/dist-es/commands/CreateCoipCidrCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCoipCidrRequestFilterSensitiveLog, CreateCoipCidrResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_ec2CreateCoipCidrCommand, serializeAws_ec2CreateCoipCidrCommand } from "../protocols/Aws_ec2";
var CreateCoipCidrCommand = (function (_super) {
    __extends(CreateCoipCidrCommand, _super);
    function CreateCoipCidrCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCoipCidrCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "CreateCoipCidrCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCoipCidrRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateCoipCidrResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCoipCidrCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2CreateCoipCidrCommand(input, context);
    };
    CreateCoipCidrCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2CreateCoipCidrCommand(output, context);
    };
    return CreateCoipCidrCommand;
}($Command));
export { CreateCoipCidrCommand };
