// @aws-sdk/client-ec2/dist-es/commands/CreateLocalGatewayRouteTableCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateLocalGatewayRouteTableRequestFilterSensitiveLog, CreateLocalGatewayRouteTableResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_ec2CreateLocalGatewayRouteTableCommand, serializeAws_ec2CreateLocalGatewayRouteTableCommand, } from "../protocols/Aws_ec2";
var CreateLocalGatewayRouteTableCommand = (function (_super) {
    __extends(CreateLocalGatewayRouteTableCommand, _super);
    function CreateLocalGatewayRouteTableCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateLocalGatewayRouteTableCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "CreateLocalGatewayRouteTableCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateLocalGatewayRouteTableRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateLocalGatewayRouteTableResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateLocalGatewayRouteTableCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2CreateLocalGatewayRouteTableCommand(input, context);
    };
    CreateLocalGatewayRouteTableCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2CreateLocalGatewayRouteTableCommand(output, context);
    };
    return CreateLocalGatewayRouteTableCommand;
}($Command));
export { CreateLocalGatewayRouteTableCommand };
