// @aws-sdk/client-ec2/dist-es/commands/CreateTransitGatewayPolicyTableCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateTransitGatewayPolicyTableRequestFilterSensitiveLog, CreateTransitGatewayPolicyTableResultFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_ec2CreateTransitGatewayPolicyTableCommand, serializeAws_ec2CreateTransitGatewayPolicyTableCommand, } from "../protocols/Aws_ec2";
var CreateTransitGatewayPolicyTableCommand = (function (_super) {
    __extends(CreateTransitGatewayPolicyTableCommand, _super);
    function CreateTransitGatewayPolicyTableCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateTransitGatewayPolicyTableCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "CreateTransitGatewayPolicyTableCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateTransitGatewayPolicyTableRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateTransitGatewayPolicyTableResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateTransitGatewayPolicyTableCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2CreateTransitGatewayPolicyTableCommand(input, context);
    };
    CreateTransitGatewayPolicyTableCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2CreateTransitGatewayPolicyTableCommand(output, context);
    };
    return CreateTransitGatewayPolicyTableCommand;
}($Command));
export { CreateTransitGatewayPolicyTableCommand };
