// @aws-sdk/client-ec2/dist-es/commands/DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationRequestFilterSensitiveLog, DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationResultFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_ec2DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand, serializeAws_ec2DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand, } from "../protocols/Aws_ec2";
var DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand = (function (_super) {
    __extends(DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand, _super);
    function DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand(input, context);
    };
    DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand(output, context);
    };
    return DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand;
}($Command));
export { DeleteLocalGatewayRouteTableVirtualInterfaceGroupAssociationCommand };
