// @aws-sdk/client-ec2/dist-es/commands/DeleteCoipCidrCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteCoipCidrRequestFilterSensitiveLog, DeleteCoipCidrResultFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_ec2DeleteCoipCidrCommand, serializeAws_ec2DeleteCoipCidrCommand } from "../protocols/Aws_ec2";
var DeleteCoipCidrCommand = (function (_super) {
    __extends(DeleteCoipCidrCommand, _super);
    function DeleteCoipCidrCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteCoipCidrCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DeleteCoipCidrCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCoipCidrRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteCoipCidrResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCoipCidrCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DeleteCoipCidrCommand(input, context);
    };
    DeleteCoipCidrCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DeleteCoipCidrCommand(output, context);
    };
    return DeleteCoipCidrCommand;
}($Command));
export { DeleteCoipCidrCommand };
