// @aws-sdk/client-ec2/dist-es/commands/DeleteCoipPoolCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteCoipPoolRequestFilterSensitiveLog, DeleteCoipPoolResultFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_ec2DeleteCoipPoolCommand, serializeAws_ec2DeleteCoipPoolCommand } from "../protocols/Aws_ec2";
var DeleteCoipPoolCommand = (function (_super) {
    __extends(DeleteCoipPoolCommand, _super);
    function DeleteCoipPoolCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteCoipPoolCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DeleteCoipPoolCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteCoipPoolRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteCoipPoolResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteCoipPoolCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DeleteCoipPoolCommand(input, context);
    };
    DeleteCoipPoolCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DeleteCoipPoolCommand(output, context);
    };
    return DeleteCoipPoolCommand;
}($Command));
export { DeleteCoipPoolCommand };
