// @aws-sdk/client-ec2/dist-es/commands/DescribeTransitGatewayRouteTableAnnouncementsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeTransitGatewayRouteTableAnnouncementsRequestFilterSensitiveLog, DescribeTransitGatewayRouteTableAnnouncementsResultFilterSensitiveLog, } from "../models/models_4";
import { deserializeAws_ec2DescribeTransitGatewayRouteTableAnnouncementsCommand, serializeAws_ec2DescribeTransitGatewayRouteTableAnnouncementsCommand, } from "../protocols/Aws_ec2";
var DescribeTransitGatewayRouteTableAnnouncementsCommand = (function (_super) {
    __extends(DescribeTransitGatewayRouteTableAnnouncementsCommand, _super);
    function DescribeTransitGatewayRouteTableAnnouncementsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeTransitGatewayRouteTableAnnouncementsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DescribeTransitGatewayRouteTableAnnouncementsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeTransitGatewayRouteTableAnnouncementsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeTransitGatewayRouteTableAnnouncementsResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeTransitGatewayRouteTableAnnouncementsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DescribeTransitGatewayRouteTableAnnouncementsCommand(input, context);
    };
    DescribeTransitGatewayRouteTableAnnouncementsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DescribeTransitGatewayRouteTableAnnouncementsCommand(output, context);
    };
    return DescribeTransitGatewayRouteTableAnnouncementsCommand;
}($Command));
export { DescribeTransitGatewayRouteTableAnnouncementsCommand };
