// @aws-sdk/client-ec2/dist-es/commands/DescribeTransitGatewayPolicyTablesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeTransitGatewayPolicyTablesRequestFilterSensitiveLog, DescribeTransitGatewayPolicyTablesResultFilterSensitiveLog, } from "../models/models_4";
import { deserializeAws_ec2DescribeTransitGatewayPolicyTablesCommand, serializeAws_ec2DescribeTransitGatewayPolicyTablesCommand, } from "../protocols/Aws_ec2";
var DescribeTransitGatewayPolicyTablesCommand = (function (_super) {
    __extends(DescribeTransitGatewayPolicyTablesCommand, _super);
    function DescribeTransitGatewayPolicyTablesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeTransitGatewayPolicyTablesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "DescribeTransitGatewayPolicyTablesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeTransitGatewayPolicyTablesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeTransitGatewayPolicyTablesResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeTransitGatewayPolicyTablesCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2DescribeTransitGatewayPolicyTablesCommand(input, context);
    };
    DescribeTransitGatewayPolicyTablesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2DescribeTransitGatewayPolicyTablesCommand(output, context);
    };
    return DescribeTransitGatewayPolicyTablesCommand;
}($Command));
export { DescribeTransitGatewayPolicyTablesCommand };
