// @aws-sdk/client-ec2/dist-es/commands/GetTransitGatewayPolicyTableAssociationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTransitGatewayPolicyTableAssociationsRequestFilterSensitiveLog, GetTransitGatewayPolicyTableAssociationsResultFilterSensitiveLog, } from "../models/models_5";
import { deserializeAws_ec2GetTransitGatewayPolicyTableAssociationsCommand, serializeAws_ec2GetTransitGatewayPolicyTableAssociationsCommand, } from "../protocols/Aws_ec2";
var GetTransitGatewayPolicyTableAssociationsCommand = (function (_super) {
    __extends(GetTransitGatewayPolicyTableAssociationsCommand, _super);
    function GetTransitGatewayPolicyTableAssociationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetTransitGatewayPolicyTableAssociationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "GetTransitGatewayPolicyTableAssociationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTransitGatewayPolicyTableAssociationsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GetTransitGatewayPolicyTableAssociationsResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTransitGatewayPolicyTableAssociationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2GetTransitGatewayPolicyTableAssociationsCommand(input, context);
    };
    GetTransitGatewayPolicyTableAssociationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2GetTransitGatewayPolicyTableAssociationsCommand(output, context);
    };
    return GetTransitGatewayPolicyTableAssociationsCommand;
}($Command));
export { GetTransitGatewayPolicyTableAssociationsCommand };
