// @aws-sdk/client-ec2/dist-es/commands/RegisterInstanceEventNotificationAttributesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RegisterInstanceEventNotificationAttributesRequestFilterSensitiveLog, RegisterInstanceEventNotificationAttributesResultFilterSensitiveLog, } from "../models/models_6";
import { deserializeAws_ec2RegisterInstanceEventNotificationAttributesCommand, serializeAws_ec2RegisterInstanceEventNotificationAttributesCommand, } from "../protocols/Aws_ec2";
var RegisterInstanceEventNotificationAttributesCommand = (function (_super) {
    __extends(RegisterInstanceEventNotificationAttributesCommand, _super);
    function RegisterInstanceEventNotificationAttributesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RegisterInstanceEventNotificationAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "EC2Client";
        var commandName = "RegisterInstanceEventNotificationAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RegisterInstanceEventNotificationAttributesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: RegisterInstanceEventNotificationAttributesResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RegisterInstanceEventNotificationAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_ec2RegisterInstanceEventNotificationAttributesCommand(input, context);
    };
    RegisterInstanceEventNotificationAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_ec2RegisterInstanceEventNotificationAttributesCommand(output, context);
    };
    return RegisterInstanceEventNotificationAttributesCommand;
}($Command));
export { RegisterInstanceEventNotificationAttributesCommand };
